gobblin-api/src/main/java/org/apache/gobblin/password/PasswordManager.java

```diff
@@ -64,34 +70,42 @@
   private static final long CACHE_SIZE = 100;
   private static final long CACHE_EXPIRATION_MIN = 10;
   private static final Pattern PASSWORD_PATTERN = Pattern.compile("ENC\\((.*)\\)");
+  private int NUMBER_OF_ENCRYPT_KEYS;
+  private FileSystem fs;
+  private String masterPasswordFile;
+  private boolean useStrongEncryptor;
 
-  private static final LoadingCache<Map.Entry<Optional<String>, Boolean>, PasswordManager> CACHED_INSTANCES =
+  private static final LoadingCache<Map.Entry<State, Boolean>, PasswordManager> CACHED_INSTANCES =
```

Comment:
Should key by a class holding the specific configuration since the state might not be equivalent even thought the same master key config is used. You need to extract out the master key path, fs uri, num keys and only use those in the key.

Reply:
done